apiserver/plane/bgtasks/work_item_link_task.py

```diff
@@ -0,0 +1,158 @@
+        }
+
+        # Fetch the main page
+        response = requests.get(url, headers=headers, timeout=10)
+        response.raise_for_status()
+
+        # Parse HTML
+        soup = BeautifulSoup(response.content, "html.parser")
+
+        # Extract title
+        title_tag = soup.find("title")
+        title = title_tag.get_text().strip() if title_tag else "No title found"
+
+        # Find favicon URL
+        favicon_url = find_favicon_url(soup, url)
+
+        # Fetch and encode favicon
+        favicon_base64 = (
+            fetch_and_encode_favicon(favicon_url, headers)
+            if favicon_url
+            else DEFAULT_FAVICON
+        )
+
+        # Prepare result
+        result = {
+            "title": title,
+            "favicon": favicon_base64,
+            "url": url,
+            "favicon_url": favicon_url,
+        }
+
+        return json.dumps(result, indent=2)
+
+    except requests.RequestException as e:
+        return json.dumps(
+            {
+                "error": f"Request failed: {str(e)}",
+                "title": None,
+                "favicon": None,
+                "url": url,
+            },
+            indent=2,
+        )
+    except Exception as e:
+        return json.dumps(
+            {
+                "error": f"Unexpected error: {str(e)}",
+                "title": None,
+                "favicon": None,
+                "url": url,
+            },
+            indent=2,
+        )
+
+
+def find_favicon_url(soup, base_url):
+    """
+    Find the favicon URL from HTML soup.
+
+    Args:
+        soup: BeautifulSoup object
+        base_url: Base URL for resolving relative paths
+
+    Returns:
+        str: Absolute URL to favicon or None
+    """
+    # Look for various favicon link tags
+    favicon_selectors = [
+        'link[rel="icon"]',
+        'link[rel="shortcut icon"]',
+        'link[rel="apple-touch-icon"]',
+        'link[rel="apple-touch-icon-precomposed"]',
+    ]
+
+    for selector in favicon_selectors:
+        favicon_tag = soup.select_one(selector)
+        if favicon_tag and favicon_tag.get("href"):
+            return urljoin(base_url, favicon_tag["href"])
+
+    # Fallback to /favicon.ico
+    parsed_url = urlparse(base_url)
+    fallback_url = f"{parsed_url.scheme}://{parsed_url.netloc}/favicon.ico"
+
+    # Check if fallback exists
+    try:
+        response = requests.head(fallback_url, timeout=5)
+        if response.status_code == 200:
+            return fallback_url
+    except Exception:
+        return DEFAULT_FAVICON
+
+    return DEFAULT_FAVICON
+
+
+def fetch_and_encode_favicon(favicon_url, headers):
+    """
+    Fetch favicon and encode it as base64.
+
+    Args:
+        favicon_url: URL to the favicon
+        headers: Request headers
+
+    Returns:
+        str: Base64 encoded favicon with data URI prefix or None
+    """
+    try:
+        response = requests.get(favicon_url, headers=headers, timeout=10)
```

Comment:
@sangeethailango Call the `find_favicon_url` from this funciton from here instead of crawl_work_item_link_title_and_favicon